app/Observers/CampaignObserver.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Observers;
+
+use App\Models\Campaign;
+
+class CampaignObserver
+{
+    /**
+     * Handle the Campaign "created" event.
+     *
+     * @param  \App\Models\Campaign  $campaign
+     * @return void
+     */
+    public function creating(Campaign $campaign)
+    {
+        $campaign->code = $campaign->generateCode();
```

Comment:
We should probably check if another campaign exists that already has that code before assigning it. Could be a simple while loop until a unique code is generated. The probability of having a collision between two campaign codes is probably astronomically low, but it's still probably a good idea to be sure.

Reply:
Hah! we actually had a review offline in which we removed exactly that functionality; since we're using `microtime()` to generate the code then 2 users would have to create a campaign at the exact same microsecond to have a collision, i'd say that for uniqueness that's acceptable. I wanted to save on having to do more queries than necessary.

Reply:
okay sounds good!